vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/KeyboardActions.java

```diff
@@ -44,6 +60,17 @@ public KeyboardActions(IWebDriverProvider webDriverProvider)
     public void typeText(WebElement element, String text)
     {
         element.sendKeys(text);
+        // https://github.com/appium/WebDriverAgent/blob/master/WebDriverAgentLib/Commands/FBCustomCommands.m#L107
+        if (genericWebDriverManager.isIOSNativeApp() && realDevice)
+        {
+            Locator keyboardReturnLocator = new Locator(AppiumLocatorType.XPATH, new SearchParameters(
+                    "//XCUIElementTypeKeyboard//XCUIElementTypeButton[@name='Return']", Visibility.VISIBLE, false));
+            searchActions.findElement(keyboardReturnLocator).ifPresentOrElse(touchActions::tap, () ->
+            {
+                throw new IllegalStateException("Unable to find 'Return' button to close the keyboard");
+            });
+            return;
+        }
         webDriverProvider.getUnwrapped(HidesKeyboard.class).hideKeyboard();
```

Comment:
if (genericWebDriverManager.isIOSNativeApp() && realDevice)
{
   ...
   ~return;~
}
else
{
        webDriverProvider.getUnwrapped(HidesKeyboard.class).hideKeyboard();
}

Reply:
Fixed, but why not to reduce number of lines by using return?